spectral_cube/io/casa_wcs.py

```diff
@@ -14,24 +60,142 @@ def wcs_casa2astropy(ndim, coordsys):
     # to CASA by getting it to write out a FITS file and reading back in
     # using WCS
 
-    try:
-        from casatools import image
-    except ImportError:
-        try:
-            from taskinit import iatool as image
-        except ImportError:
-            raise ImportError("Could not import CASA (casac) and therefore cannot convert csys to WCS")
-
-    tmpimagefile = tempfile.mktemp() + '.image'
-    tmpfitsfile = tempfile.mktemp() + '.fits'
-    ia = image()
-    ia.fromarray(outfile=tmpimagefile,
-                 pixels=np.ones([1] * ndim),
-                 csys=coordsys, log=False)
-    ia.done()
-
-    ia.open(tmpimagefile)
-    ia.tofits(tmpfitsfile, stokeslast=False)
-    ia.done()
-
-    return WCS(tmpfitsfile)
+    header = fits.Header()
+
+    # Observer information (ObsInfo.cc)
+
+    header['OBSERVER'] = coordsys['observer']
+    header['TELESCOP'] = coordsys['telescope']
+
+    if coordsys['obsdate']['refer'] == 'LAST':
+
+        header['TIMESYS'] = 'UTC'
+
+    else:
+
+        header['TIMESYS'] = coordsys['obsdate']['refer']
+        header['MJD-OBS'] = coordsys['obsdate']['m0']['value']
+
+        dateobs = Time(header['MJD-OBS'],
+                       format='mjd',
+                       scale=coordsys['obsdate']['refer'].lower())
+        dateobs.precision = 6
+        header['DATE-OBS'] = dateobs.isot
+
+    if 'telescopeposition' in coordsys:
+
+        obsgeo_lon = coordsys['telescopeposition']['m0']['value']
+        obsgeo_lat = coordsys['telescopeposition']['m1']['value']
+        obsgeo_alt = coordsys['telescopeposition']['m2']['value']
+
+        header['OBSGEO-X'] = obsgeo_alt * np.cos(obsgeo_lon) * np.cos(obsgeo_lat)
+        header['OBSGEO-Y'] = obsgeo_alt * np.sin(obsgeo_lon) * np.cos(obsgeo_lat)
+        header['OBSGEO-Z'] = obsgeo_alt * np.sin(obsgeo_lat)
+
+    # World coordinates
+
+    # Find worldmap entries
+
+    worldmap = {}
+
+    for key, value in coordsys.items():
+        if key.startswith('worldmap'):
+            index = int(key[8:])
+            worldmap[index] = value
+
+    # Now iterate through the different coordinate types to populate the WCS
+
+    header['WCSAXES'] = np.max([np.max(idx) + 1 for idx in worldmap.values()])
+
+    # Initialize PC
+    for i in range(header['WCSAXES']):
+        for j in range(header['WCSAXES']):
+            header[f'PC{i+1}_{j+1}'] = 0.
+
+    for coord_type in ('direction', 'spectral', 'stokes', 'linear'):
+
+        for index in range(len(worldmap)):
+            if f'{coord_type}{index}' in coordsys:
+                break
+        else:
+            continue
```

Comment:
Just commenting that I have never before seen the `for/else` used in practice.    It's uncommon enough that I would suggest including a comment about it; at first I was convinced this was a typo.

Reply:
I've added a comment. The ``else`` for a ``for`` loop is quite useful for situations like this!